<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Database Query Builder</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <widget class="QWidget" name="tabQuery">
       <attribute name="title">
        <string>Query Builder</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="gbConnection">
          <property name="title">
           <string>Connection Settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QRadioButton" name="rbSQLite">
             <property name="text">
              <string>SQLite</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="lblSQLitePath">
               <property name="text">
                <string>Database path:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="leSQLitePath"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QRadioButton" name="rbPostgreSQL">
             <property name="text">
              <string>PostgreSQL</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="lblPGHost">
               <property name="text">
                <string>Host:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="lePGHost"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="lblPGDatabase">
               <property name="text">
                <string>Database:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="lePGDatabase"/>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="lblPGUser">
               <property name="text">
                <string>User:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="lePGUser"/>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="lblPGPassword">
               <property name="text">
                <string>Password:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLineEdit" name="lePGPassword">
               <property name="echoMode">
                <enum>QLineEdit::Password</enum>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="lblPGPort">
               <property name="text">
                <string>Port:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="sbPGPort">
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>5432</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="lblConnectionName">
               <property name="text">
                <string>Connection name:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="leConnectionName"/>
             </item>
             <item>
              <widget class="QPushButton" name="btnConnect">
               <property name="text">
                <string>Connect</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnDisconnect">
               <property name="text">
                <string>Disconnect</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbExplorer">
          <property name="title">
           <string>Database Explorer</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="lblConnections">
               <property name="text">
                <string>Connection:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cbConnections"/>
             </item>
             <item>
              <widget class="QLabel" name="lblTables">
               <property name="text">
                <string>Table:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cbTables"/>
             </item>
             <item>
              <widget class="QPushButton" name="btnBuildQuery">
               <property name="text">
                <string>Build Query</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QTableView" name="tvResults">
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbQuery">
          <property name="title">
           <string>Query Builder</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QPlainTextEdit" name="pteQuery"/>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QPushButton" name="btnExecute">
               <property name="text">
                <string>Execute</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnClear">
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnExport">
               <property name="text">
                <string>Export to CSV</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabLog">
       <attribute name="title">
        <string>Query Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QPlainTextEdit" name="pteQueryLog">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="btnClearLog">
            <property name="text">
             <string>Clear Log</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnSaveLog">
            <property name="text">
             <string>Save Log</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>21</height>
    </rect>
   </property> 
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>